DataProcessing/process.py

```diff
@@ -0,0 +1,109 @@
+
+for survey_name, series_ids in SERIES_IDS_BY_SURVEY_NAME.items():
+    docs_table_file.write(f"""
+<h4>{survey_name}</h4>
+<p>The following table show the supported series in the {survey_name} survey:</p>
+<table class='qc-table table'>""")
+    series_ids = sorted(series_ids)
+    survey_code = series_ids[0][:2].lower()
+    response = requests.get(f'https://download.bls.gov/pub/time.series/{survey_code}/{survey_code}.series', headers=HEADERS)
+    lines = response.text.split("\n")
+    column_names = lines[0].split("\t")
+    column_names = [n.strip() for n in column_names if n != ""] # Handles the case where there can be a column names that's empty (survey CF is an example)
+    column_to_drop = None
+
+    series_by_series_id = {}
+    for z, line in enumerate(lines[1:-1]):
+        adj_line = line.split("\t")
+        if len(adj_line) == 0:
+            continue
+        if adj_line[-1] == "": # Handles the case in the HS survey where there is an extra column at the end (which is empty)
+            adj_line = adj_line[:-1]
+        adj_line = [cell.strip() for cell in adj_line if cell != "n"] # Handles the case where this row has more columns than there are column names and the column is just filled with `n` (survey EE and SA are examples of this)
+        series_id = adj_line[0]
+        if series_id not in series_ids:
+            continue
+        if adj_line[1:] == []: # Handles the case where https://download.bls.gov/pub/time.series/... doesn't exist (there are 5 surveys)
+            continue
+        if len(adj_line) != len(column_names) and column_to_drop is None: # Handles the case where the number of cells in each row is greater than the number of columns (see surveys EB, EC, GP, HS, PD, SH, and SI)
+            column_to_drop = adj_line.index("") # Drop the first empty column
+        if column_to_drop is not None:
+            del adj_line[column_to_drop]
+        
+        title = adj_line[column_names.index("series_title")]
+        begin_year = int(adj_line[column_names.index("begin_year")])
+        end_year = int(adj_line[column_names.index("end_year")])
+        series_by_series_id[series_id] = Series(series_id, begin_year, end_year, title, survey_code)
+
+    series_id_written_to_docs_file = []
+    docs_table_column_names = []
+    # Get and save time series data
+    for j in range(0, len(series_ids), MAX_SERIES_IDS): 
+        series_ids_subset = series_ids[j:j + MAX_SERIES_IDS]
+        start_year = min([series_by_series_id[series_id].begin_year for series_id in series_ids_subset])
+        end_year = max([series_by_series_id[series_id].end_year for series_id in series_ids_subset])
+
+        for start_year in range(start_year, end_year+1, MAX_TIME_FRAME_YEARS): 
+            data = {
+                "seriesid": series_ids_subset, 
+                "startyear": str(start_year), # inclusive
+                "endyear": str(min(start_year + MAX_TIME_FRAME_YEARS -1, end_year)), # inclusive
```

Comment:
Just checking but we regenerate the whole history each time? how long does it take?